drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -432,39 +482,38 @@ def _get_zone_by_uid(self, uid):
             if zone["uid"] == uid:
                 return zone
 
-    def _get_grade(self):
+    def _get_item_stats(self):
         """
-        Returns the student's grade for this block.
+        Returns a 3-tuple representing the number of correct items placed, the number of completed
+        items, right or wrong, and the total number of items that must be completed.
         """
-        correct_count = 0
-        total_count = 0
+        all_items = self.data['items']
         item_state = self._get_item_state()
 
-        for item in self.data['items']:
-            if item['zone'] != 'none':
-                total_count += 1
-                item_id = str(item['id'])
-                if item_id in item_state:
-                    correct_count += 1
+        required_items = [str(item['id']) for item in all_items if self._get_item_zones(item['id']) != []]
+        completed_items = [item for item in required_items if item in item_state]
```

Comment:
@haikuginger That's great that you went ahead and introduced this method :)

I think you can simplify it a bit though: Since #80, placing an item on a correct zone is equivalent to "completing" it. (Before, there was an option for course authors to define items that required a numeric input; those items were considered "completed" iff they had been placed on a correct zone *and* a numeric input had been submitted for them.) So we don't need a "completed" stat; `_get_grade` and `_is_finished` just need `correct_count` and `total_count` to produce an appropriate result.